.github/workflows/deploy-config.yml

```diff
@@ -0,0 +1,33 @@
+name: Deploy config
+on:
+  - create
+  - delete
+
+jobs:
+  build:
+    name: Deploy config
+    runs-on: ubuntu-latest
+    env:
+      S3_BUCKET: crystal-book
+    steps:
+      - name: Download source
+        uses: actions/checkout@v2
+      - name: Build versions.json
+        run: scripts/docs-versions.sh origin | tee /dev/stderr > versions.json
+      - name: Build aws-config.json
+        run: |
+          latest_version=$(git ls-remote --heads origin | grep -P -o '(?<=refs/heads/)[0-9][0-9.]+' | sort -V -r | head -n1)
```

Comment:
So now we have this `git ls-remote` code in two places. I don't understand what the trouble was to keep this code snippet within the script itself. Can I change it back?

Reply:
It's different concerns. I'd like that to be separate processes. The current implementation closely resembles the setup we're already using for the API docs over at https://github.com/crystal-lang/distribution-scripts/blob/8328ae59292c05cf8c830a34b5b39ac46c43813e/docs/Makefile#L43-L55
I'm sure we can remove the duplicate invocation by caching the result in an output variable.

Reply:
It's not different concerns, though. It is exactly the same concern - "make versions up to date", just that AWS happens to handle one side of it separately.

Reply:
Consider that both sides of this operation mainly change the meaning of "latest" - one for MkDocs and one for AWS. They shouldn't be separated.

Reply:
With this in mind, I added back some of my refactors. Sorry for the trouble and I realize it's kinda rude.
We can definitely keep discussing this and change back still.